apps/spruce/src/components/SpruceForm/Widgets/CheckboxWithConditionalBannerWidget.tsx

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect } from "react";
+import Banner from "@leafygreen-ui/banner";
+import { LeafyGreenCheckBox } from "components/SpruceForm/Widgets/LeafyGreenWidgets";
+import { SpruceWidgetProps } from "components/SpruceForm/Widgets/types";
+
+interface Props extends SpruceWidgetProps {
+  options: SpruceWidgetProps["options"] & {
+    originalValue: boolean;
+    "data-cy-banner"?: string;
+    enableCopy?: string;
+    disableCopy?: string;
+  };
+}
+export const CheckboxWithConditionalBannerWidget: React.FC<Props> = (props) => {
```

Comment:
I don't think this separate component is necessary. We already have a [Checkbox widget](https://github.com/evergreen-ci/ui/blob/a4589de0dd2de34be73ae09184dd31b8e45f4885/apps/spruce/src/components/SpruceForm/Widgets/LeafyGreenWidgets.tsx#L119). 
We also have an established pattern for warnings and errors alongside components like the one we used for the [RadioBox](https://github.com/evergreen-ci/ui/blob/a4589de0dd2de34be73ae09184dd31b8e45f4885/apps/spruce/src/components/SpruceForm/Widgets/LeafyGreenWidgets.tsx#L365). I think leveraging the validator patterns the library provides out of the box will make this more reusable.

Reply:
Heres another [example](https://github.com/evergreen-ci/ui/blob/0e6355ba4905114f1bbbe6195c02b2c48761c502/apps/spruce/src/pages/spawn/spawnHost/spawnHostButton/SpawnHostModal.tsx#L95) of alternative dynamic way to handle warnings